src/main/scala/io/iohk/ethereum/ledger/InMemoryWorldStateProxy.scala

```diff
@@ -155,7 +155,8 @@ class InMemoryWorldStateProxy private(
     copyWith(accountsStateTrie = accountsStateTrie.put(address.bytes, account))
 
   override def deleteAccount(address: Address): InMemoryWorldStateProxy =
-    copyWith(accountsStateTrie = accountsStateTrie.remove(address.bytes))
+    copyWith(accountsStateTrie = accountsStateTrie.remove(address.bytes),
+      contractStorages = contractStorages - address)
```

Comment:
what in case when contract MPT was shared with other contracts as well as EVM code?